### PR TITLE
Add Snowflake connector

### PR DESCRIPTION
**Pull request by kapilraajP**
[Snowflake](https://docs.snowflake.com/en/user-guide/jdbc.html) connector allows you to programmatically access all of the Snowflake applications, databases, APIs, services via the Java Database Connectivity (JDBC) API using Ballerina.
It provides operations to execute a wide range of standard DDL Commands, SQL Commands, and SQL Functions for querying data sources. 
You can find reference information for all the Snowflake SQL commands (DDL, DML, and query syntax) [here](https://docs.snowflake.com/en/sql-reference-commands.html).

Resolves https://github.com/ballerina-platform/ballerina-extended-library/issues/344

- Add Snowflake connector module
- Add documentation
- Add samples

## Automation tests
 - Unit tests 
   > Done
 - Integration tests
   > Done

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran the FindSecurityBugs plugin and verified the report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

Ballerina Version: 2201.0.2
Operating System: Ubuntu 20.04
Java SDK: 11

### CODE REVIEW

File: ballerina/Module.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import ballerinax/snowflake;      // Get the Snowflake connector
+import ballerinax/snowflake.driver as _;   // Get the Snowflake driver
+```
+
+### Step 2: Create a new connector instance
+Provide the `account_identifier`, `<username>` and `<password>` to initialize the Snowflake connector. 
+options should be provided as,
+```
+snowflake:Options options = {
+    requestGeneratedKeys: snowflake:NONE  // This should be specified
```
Comment (RolandHewage) [inline ballerina/Module.md:32]: Shall we add this line too in the doc
```
The `requestGeneratedKeys` option must be set to `jdbc:NONE` as snowflake does not support the retrieval of auto-generated keys.
```

File: ballerina/Module.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+        sql:ParameterizedQuery[] insertQueries = 
+            from var data in insertRecords
+            select `INSERT INTO COMPANYDB.PUBLIC.EMPLOYEES
+                    (FirstName, LastName, Company)
+                    VALUES (${data.FirstName}, ${data.LastName}, ${data.Company})`;
+
+        _ = check snowflakeClient->batchExecute(insertQueries);
+    }
+    ```
```
Comment (RolandHewage) [inline ballerina/Module.md:125]: If possible try to add a simple example for executing a stored procedure too. Just a suggestion.

File: native/src/main/java/io/ballerina/stdlib/java/jdbc/Constants.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        public static final BString OTHER = StringUtils.fromString("Other");
+        public static final BString POOL_IDLE_TIMEOUT = StringUtils.fromString("PoolIdleTimeout");
+        public static final BString POOL_MAX_SIZE = StringUtils.fromString("PoolMaxSize");
+        public static final BString POOL_MIN_SIZE = StringUtils.fromString("PoolMinSize");
+        public static final BString POOL_WAIT_TIME = StringUtils.fromString("PoolWaitTime");
+        public static final BString PSEUDO_COLUMNS = StringUtils.fromString("PseudoColumns");
+        public static final BString QUERY_PASS_THROUGH = StringUtils.fromString("QueryPassthrough");
+        public static final BString RTK = StringUtils.fromString("RTK");
+        public static final BString TIMEOUT = StringUtils.fromString("Timeout");
+        public static final BString USE_CONNECTION_POOLING = StringUtils.fromString("UseConnectionPooling");
```
Comment (RolandHewage) [inline native/src/main/java/io/ballerina/stdlib/java/jdbc/Constants.java]: Please remove these unused constants from the CData connect module.

File: native/src/main/java/io/ballerina/stdlib/java/jdbc/nativeimpl/ClientProcessor.java [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+                if (miscellaneousMap.containsKey(Constants.MiscellaneousConfig.TIMEOUT)) {
+                    properties.put(Constants.ConnectionStringProps.TIMEOUT,
+                            miscellaneousMap.getIntValue(Constants.MiscellaneousConfig.TIMEOUT));
+                }
+                if (miscellaneousMap.containsKey(Constants.MiscellaneousConfig.USE_CONNECTION_POOLING)) {
+                    properties.put(Constants.ConnectionStringProps.USE_CONNECTION_POOLING,
+                            miscellaneousMap.getBooleanValue(Constants.MiscellaneousConfig.USE_CONNECTION_POOLING));
+                }
+            }
+        }
```
Comment (RolandHewage) [inline native/src/main/java/io/ballerina/stdlib/java/jdbc/nativeimpl/ClientProcessor.java:73]: We have to handle the options based on the Snowflake API requirements. Please remove these unncessary options handled specific for the CData connect domain. You can use the approach used in the following repo https://github.com/ballerina-platform/module-ballerinax-java.jdbc/blob/master/native/src/main/java/io/ballerina/stdlib/java/jdbc/nativeimpl/ClientProcessor.java.

**Review by RolandHewage — COMMENTED** [review]
Requested few changes. Other than that the rest seems fine. Please do the changes in the native code.

File: ballerina/Module.md (hunk 2 above)
Comment (kapilraajP) [reply ballerina/Module.md:125]: Added the sample code snippet

File: native/src/main/java/io/ballerina/stdlib/java/jdbc/Constants.java (hunk 3 above)
Comment (kapilraajP) [reply native/src/main/java/io/ballerina/stdlib/java/jdbc/Constants.java]: Removed that constants

File: native/src/main/java/io/ballerina/stdlib/java/jdbc/nativeimpl/ClientProcessor.java (hunk 4 above)
Comment (kapilraajP) [reply native/src/main/java/io/ballerina/stdlib/java/jdbc/nativeimpl/ClientProcessor.java:73]: Updated

File: ballerina/Module.md (hunk 1 above)
Comment (kapilraajP) [reply ballerina/Module.md:32]: Added that too in doc

**Comment by kapilraajP** [comment]
> Requested few changes. Other than that the rest seems fine. Please do the changes in the native code.

Fix done for above mentioned review comments

**Review by RolandHewage — APPROVED** [review]

**Pull request merged**